promotion/src/main/java/com/yas/promotion/repository/PromotionRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.yas.promotion.repository;
+
+import com.yas.promotion.model.Promotion;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PromotionRepository extends JpaRepository<Promotion, Long> {
+
+    @Query("select p from Promotion p where p.name = :name and (:id is null or p.id != :id)")
```

Comment:
shouldn't select the whole entity just to check for it's existence, I suggest using select count(1) instead, it would give you 1 incase the record exists or else 0.

Reply:
thanks for this suggest, I'll change it